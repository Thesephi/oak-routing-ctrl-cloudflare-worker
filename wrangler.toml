name = "oak-routing-ctrl-cloudflare"
main = "src/index.ts"
compatibility_date = "2024-06-17"
workers_dev = true
