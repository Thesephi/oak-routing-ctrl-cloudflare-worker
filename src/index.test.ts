import { unstable_dev } from "wrangler";
import type { UnstableDevWorker } from "wrangler";
import { describe, expect, it, beforeAll, afterAll } from "vitest";

describe("oak-routing-ctrl-cloudflare", () => {
  let worker: UnstableDevWorker;

  beforeAll(async () => {
    worker = await unstable_dev("src/index.ts", { experimental: { disableExperimentalWarning: true } });
  });

  afterAll(async () => {
    await worker.stop();
  });

  it("should return 200 response with expected data", async () => {
    const resp = await worker.fetch(`/echo/tester`, {
      method: "post",
      headers: { "content-type": "application/json" },
      body: JSON.stringify({ foo: "bar" }),
    });
    expect(resp.status).toBe(200);

    const respData = await resp.json();
    expect(respData).toMatchObject({
      body: { foo: "bar" },
      param: { name: "tester" },
    });
  });

  it("should serve Open API Spec", async () => {
    const resp = await worker.fetch(`/oas.json`);
    expect(resp.status).toBe(200);

    const respData = await resp.json();
    expect(respData).toMatchSnapshot();
  });
});
