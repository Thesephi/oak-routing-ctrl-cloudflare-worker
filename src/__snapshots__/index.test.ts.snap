// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`oak-routing-ctrl-cloudflare > should serve Open API Spec 1`] = `
{
  "components": {
    "parameters": {},
    "schemas": {},
  },
  "info": {
    "description": "This Open API Spec is served by a Cloudflare Workers module. To see the raw spec json, simply request at \`/oas.json\`",
    "title": "Demo API service powered by Cloudflare Workers",
    "version": "1.0.0",
  },
  "openapi": "3.0.0",
  "paths": {
    "/echo/{name}": {
      "get": {
        "parameters": [
          {
            "in": "path",
            "name": "name",
            "required": true,
            "schema": {
              "type": "string",
            },
          },
        ],
        "responses": {
          "200": {
            "content": {
              "application/json": {
                "schema": {
                  "properties": {
                    "body": {
                      "properties": {},
                      "type": "object",
                    },
                    "headers": {
                      "properties": {},
                      "type": "object",
                    },
                    "param": {
                      "properties": {},
                      "type": "object",
                    },
                    "query": {
                      "properties": {},
                      "type": "object",
                    },
                  },
                  "required": [
                    "query",
                    "body",
                    "param",
                    "headers",
                  ],
                  "type": "object",
                },
              },
            },
            "description": "Success response",
          },
        },
      },
      "post": {
        "parameters": [
          {
            "in": "path",
            "name": "name",
            "required": true,
            "schema": {
              "type": "string",
            },
          },
        ],
        "responses": {
          "200": {
            "content": {
              "application/json": {
                "schema": {
                  "properties": {
                    "body": {
                      "properties": {},
                      "type": "object",
                    },
                    "headers": {
                      "properties": {},
                      "type": "object",
                    },
                    "param": {
                      "properties": {},
                      "type": "object",
                    },
                    "query": {
                      "properties": {},
                      "type": "object",
                    },
                  },
                  "required": [
                    "query",
                    "body",
                    "param",
                    "headers",
                  ],
                  "type": "object",
                },
              },
            },
            "description": "Success response",
          },
        },
      },
    },
  },
  "servers": [
    {
      "url": "/",
    },
  ],
}
`;
